contracts/contribute/ContributeManager.sol

```diff
@@ -180,6 +196,37 @@ contract ContributeManager is ERC721TokenReceiver {
         _nonce = localNonce + 1;
     }
 
+    function _reDeploy(uint256 serviceId, address stakingInstance, address multisig) internal {
```

Comment:
This enables to re-deploy the service that is in the pre-registration state.

Reply:
Presumably this is the same logic as initial deploy? Why are we not re-using logic?

Reply:
The previous one happened to be incomplete.

Reply:
Please unify in the end with old logic so we have less duplicate code.